### DISCUSSION TITLE
Hi all. I have two very big problems. Metamask connector&wallet connect

### DISCUSSION THREAD

**Discussion by SobeitB** [wevm/wagmi#2246] [Q&A]
1. I haven't changed anything, but the metamask connector gives an error
![image](https://user-images.githubusercontent.com/91097143/235321135-070298f4-96cb-4b6b-8c8f-561f1360ec87.png)
Sometimes it is an error that the user rejected the connection request
2. 2. Web3Modal does not display the metamask in its modal window. How can I control it?

**Comment by tmm** [comment]
Please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). It's hard to help out without being able to reproduce what you are seeing. For your second question, you might want to ask over on the [Web3Modal](https://github.com/WalletConnect/web3modal) repo.

**Reply by SobeitB** [reply]
1. provider
![image](https://user-images.githubusercontent.com/91097143/235321700-82cdc704-f59a-48e7-b8f2-8df066821d81.png)
2. connectorMetamask
const connectorMetamask = new MetaMaskConnector({
    chains,
})
3. wagmi client
![image](https://user-images.githubusercontent.com/91097143/235321730-12ef6e7c-53d0-4400-a37c-c2531735ee8a.png)

4. connect: await connect({ connector: connectorMetamask })
